src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -4990,29 +4990,26 @@ namespace detail
         return fabs((a - d).dot((b - d).cross(c - d))) / 6.0f;
     }
 
-    // T is the type of 'tri_to_poly' values
-    // U is the type of connectivity values
-    // V is the type of coordset values
-    template<typename T, typename U, typename V>
-    void
+    // determines the type of the coordinate values and calls
     // we want access to the new topology so we can calculate the areas
     // of the new triangles/volumes of the new tetrahedra
-    volume_dependent_helper(const Node &topo_dest,
-                            const Node &coordset_dest,
-                            const int dimensions,
-                            const int new_num_shapes, // number of new triangles or tetrahedrons
-                            const int num_orig_shapes, // number of original polygons or polyhedra
-                            const T *tri_to_poly,
-                            Node &volumes_info,
-                            Node &volumes_field_values)
+    void
+    volume_dependent(const Node &topo_dest,
+                     const Node &coordset_dest,
+                     const int dimensions,
+                     const int new_num_shapes, // number of new triangles or tetrahedrons
+                     const int num_orig_shapes, // number of original polygons or polyhedra
+                     int_accessor tri_to_poly,
+                     Node &volumes_info,
+                     Node &volumes_field_values)
     {
         // first we calculate the volume of each triangle
         volumes_field_values.set(conduit::DataType::float64(new_num_shapes));
-        float64 *tri_volumes = volumes_field_values.value();
+        float64_array tri_volumes = volumes_field_values.value();
 
-        const U *connec = topo_dest["elements/connectivity"].value();
-        const V *coords_x = coordset_dest["values/x"].value();
-        const V *coords_y = coordset_dest["values/y"].value();
+        int_accessor connec = topo_dest["elements/connectivity"].value();
```

Comment:
to be safe, i think these guys should be `int64_accessor` and `float64_accessor`